docs/configs-microbenchmark-test/run_config1.py

```diff
@@ -0,0 +1,222 @@
+               SIMD_Unit(), WritePort(), RdWrPort(), IprPort() ]
+
+
+class O3_W256CPU(DerivO3CPU):
+    branchPred = BranchPredictor()
+    fuPool = Ideal_FUPool()
+    fetchWidth = 32
+    decodeWidth = 32
+    renameWidth = 32
+    dispatchWidth = 32
+    issueWidth = 32
+    wbWidth = 32
+    commitWidth = 32
+    squashWidth = 32
+    fetchQueueSize = 256
+    LQEntries = 250
+    SQEntries = 250
+    numPhysIntRegs = 256
+    numPhysFloatRegs = 256
+    numIQEntries = 256
+    numROBEntries = 256
+
+
+class O3_W2KCPU(DerivO3CPU):
+    branchPred = BranchPredictor()
+    fuPool = Ideal_FUPool()
+    fetchWidth = 32
+    decodeWidth = 32
+    renameWidth = 32
+    dispatchWidth = 32
+    issueWidth = 32
+    wbWidth = 32
+    commitWidth = 32
+    squashWidth = 32
+    fetchQueueSize = 256
+    LQEntries = 250
+    SQEntries = 250
+    numPhysIntRegs = 1024
+    numPhysFloatRegs = 1024
+    numIQEntries = 2096
+    numROBEntries = 2096
+
+class SimpleCPU(TimingSimpleCPU):
+    branchPred = BranchPredictor()
+
+class DefaultO3CPU(DerivO3CPU):
+    branchPred = BranchPredictor()
+
+
+
+
+# Add more CPUs under test before this
+valid_cpus = [SimpleCPU, DefaultO3CPU, O3_W256CPU, O3_W2KCPU]
+
+valid_cpus = {cls.__name__[:-3]:cls for cls in valid_cpus}
+
+class InfMemory(SimpleMemory):
+    latency = '0ns'
+    bandwidth = '0B/s'
+
+class SingleCycleMemory(SimpleMemory):
+    latency = '1ns'
+    bandwidth = '0B/s'
+
+class SlowMemory(SimpleMemory):
+    latency = '100ns'
+    bandwidth = '0B/s'
+
+# Add more Memories under test before this
+valid_memories = [InfMemory, SingleCycleMemory, SlowMemory]
+valid_memories = {cls.__name__[:-6]:cls for cls in valid_memories}
+parser = argparse.ArgumentParser()
+parser.add_argument('cpu', choices = valid_cpus.keys())
+parser.add_argument('memory_model', choices = valid_memories.keys())
+parser.add_argument('binary', type = str, help = "Path to binary to run")
+parser.add_argument('--l1i_size',type = str,  help = "CachesizeofL1i")
+parser.add_argument('--l1d_size',type = str,  help = "CachesizeofL1d")
+parser.add_argument('--l2_size', type = str,  help = "CachesizeofL2")
+
+args  = parser.parse_args()
+
+    #parser.add_argument('binary_input', type = str, help = "Inputs to the binary")
+
+command_line_args(args.l1i_size,args.l1d_size,args.l2_size)
+
+class MySystem(BaseTestSystem):
+    _CPUModel = valid_cpus[args.cpu]
+    _MemoryModel = None
+system = MySystem()
+system.setTestBinary(args.binary)
+root = Root(full_system = False, system = system)
+m5.instantiate()
+
+exit_event = m5.simulate()
+
+if exit_event.getCause() != 'exiting with last active thread context':
+    print("Benchmark failed with bad exit cause.")
+    print(exit_event.getCause())
+    exit(1)
+if exit_event.getCode() != 0:
+    print("Benchmark failed with bad exit code.")
+    print("Exit code {}".format(exit_event.getCode()))
+    exit(1)
+
+print("{} ms".format(m5.curTick()/1e9))
```

Comment:
There should be a function in `m5` which will do this conversion from ticks to ms. Using 1e9 could be wrong if the tick rate changed (this won't happen in practice, but it theoretically possible).